Gamepads as entities release notes (#1762)
bevyengine/bevy-website: release-content/0.15/release-notes/12770_Implement_gamepads_as_entities.md

<!-- Implement gamepads as entities -->
<!-- https://github.com/bevyengine/bevy/pull/12770 -->

> Gamepads are surprisingly complex objects: each with their own array of settings and unique identifier.

Reviewer: ```suggestion
Gamepads are surprisingly complex objects: each with their own array of settings and unique identifiers.
```

> To make it easier to work with gamepads, each gamepad is now a distinct entity.

Reviewer: ```suggestion
To make them easier to work with, each gamepad is now a distinct entity.
```

> Metadata and the state of each button and axis is stored on the [`Gamepad`] component, and gamepad-specific settings like deadzones and sensitivity can be configured with the [`GamepadSettings`] component.

Reviewer: ```suggestion
Metadata, the state of each button and axis is stored on the [`Gamepad`] component, while gamepad-specific settings like deadzones and sensitivity can be configured with the [`GamepadSettings`] component.
```

           .get(GamepadButton::new(
               gamepad,
               GamepadButtonType::RightTrigger2,
           ))
           .unwrap();
        if right_trigger.abs() > 0.01 {
            info!("RightTrigger2 value is {}", right_trigger);      
        }

        let left_stick_x = axes
           .get(GamepadAxis::new(gamepad, GamepadAxisType::LeftStickX))
           .unwrap();
        if left_stick_x.abs() > 0.01 {
            info!("LeftStickX value is {}", left_stick_x);        
        }
    }
}
```

In 0.15, we can write this much more simply as:

```rust
fn gamepad_system(
   gamepads: Query<&Gamepad>
) {
    for gamepad in gamepads.iter() {
      if gamepad.just_pressed(GamepadButton::South) {
            println!("just pressed South");
        } 
         
        let right_trigger = gamepad.get(GamepadButton::RightTrigger2).unwrap();
        if right_trigger.abs() > 0.01 {
            info!("RightTrigger2 value is {}", right_trigger);      
        }

        let left_stick_x = gamepad.get(GamepadAxis::LeftStickX).unwrap();
        if left_stick_x.abs() > 0.01 {
            info!("LeftStickX value is {}", left_stick_x);        
        }
    }
}
```

> This design makes adding custom behavior to each gamepad easy, and  opens the door to supporting more advanced features such as rumble and gyroscopic control directly in Bevy in the future!

Reviewer: ```suggestion
Not only does this design make gamepad-specific custom behavior easy, it also opens the door to supporting more advanced features such as rumble and gyroscopic control in the future!
```

[`Gamepad`]: https://docs.rs/bevy/0.15.0-rc.2/bevy/input/gamepad/struct.Gamepad.html

> [`GamepadSettings`]: https://docs.rs/bevy/0.15.0-rc.2/bevy/input/gamepad/struct.GamepadSettings.html

Reviewer: I need to add this to fallible param notes
